lib/chef/http/basic_client.rb

```diff
@@ -95,26 +95,8 @@ def request(method, url, req_body, base_headers = {})
         raise
       end
 
-      #adapted from buildr/lib/buildr/core/transports.rb
       def proxy_uri
-        proxy = Chef::Config["#{url.scheme}_proxy"] ||
-          env["#{url.scheme.upcase}_PROXY"] || env["#{url.scheme}_proxy"]
-
-        # Check if the proxy string contains a scheme. If not, add the url's scheme to the
-        # proxy before parsing. The regex /^.*:\/\// matches, for example, http://. Reusing proxy
-        # here since we are really just trying to get the string built correctly.
-        if String === proxy && !proxy.strip.empty?
-          if proxy.match(/^.*:\/\//)
-            proxy = URI.parse(proxy.strip)
-          else
-            proxy = URI.parse("#{url.scheme}://#{proxy.strip}")
-          end
-        end
-
-        no_proxy = Chef::Config[:no_proxy] || env["NO_PROXY"] || env["no_proxy"]
-        excludes = no_proxy.to_s.split(/\s*,\s*/).compact
-        excludes = excludes.map { |exclude| exclude =~ /:\d+$/ ? exclude : "#{exclude}:*" }
-        return proxy unless excludes.any? { |exclude| File.fnmatch(exclude, "#{host}:#{port}") }
+        @proxy_uri ||= ChefConfig::Config.proxy_uri(url.scheme, host, port)
```

Comment:
Not sure I understand: the [signature](https://github.com/chef/chef/pull/4348/files#diff-d19a42ae266a2919d7cfeb43c77dbaffR843) for `ChefConfig::Config.proxy_uri` is

```ruby
def self.proxy_uri(scheme, host, port)
```